<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">



<title>Investigating a pseudo-horseshoe prior for the MR-Egger model</title>

<script src="mr2019-okezie-poster_files/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="mr2019-okezie-poster_files/viz-0.3/viz.js"></script>
<link href="mr2019-okezie-poster_files/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="mr2019-okezie-poster_files/grViz-binding-1.0.1/grViz.js"></script>
<script src="mr2019-okezie-poster_files/kePrint-0.0.1/kePrint.js"></script>





<!--
Font-awesome icons ie github or twitter
-->
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/all.css" integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf" crossorigin="anonymous">
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/brands.css" integrity="sha384-n9+6/aSqa9lBidZMRCQHTHKJscPq6NW4pCQBiMmHdUCvPN8ZOg2zJJTkC7WIezWv" crossorigin="anonymous">

<!--
Google fonts api stuff
-->
<link href='https://fonts.googleapis.com/css?family=Special Elite' rel='stylesheet'>
<link href='https://fonts.googleapis.com/css?family=Rasa' rel='stylesheet'>

<!--
Here are the required style attributes for css to make this poster work :)
-->
<style>
@page {
size: 33.1in 46.8in;
margin: 0;
padding: 0;
}
body {
margin: 0px;
padding: 0px;
width: 33.1in;
height: 46.8in;
text-align: justify;
font-size: 45px;
line-height: 1.05;
}
/* RMarkdown Class Styles */
/* center align leaflet map,
from https://stackoverflow.com/questions/52112119/center-leaflet-in-a-rmarkdown-document */
.html-widget {
margin: auto;
position: sticky;
margin-top: 2cm;
margin-bottom: 2cm;
}
.leaflet.html-widget.html-widget-static-bound.leaflet-container.leaflet-touch.leaflet-fade-anim.leaflet-grab.leaflet-touch-drag.leaflet-touch-zoom {
position: sticky;
width: 100%;
}
pre.sourceCode.r {
background-color: #dddddd40;
border-radius: 4mm;
padding: 4mm;
width: 75%;
margin: auto;
margin-top: 1em;
margin-bottom: 1em;
/* align-items: center; */
}
code.sourceCode.r{
background-color: transparent;
font-size: 20pt;
/* border-radius: 2mm;
padding: 2mm; */
}
code {
font-size: 20pt;
font-family: monospace;
background-color: #B5121B24;
color: #B5121B;
padding: 1.2mm;
line-height: 1;
border-radius: 2mm;
}
caption {
margin-bottom: 10px;
font-size: 20pt;
font-style: italic;
}

tbody tr:nth-child(odd) {
    background-color: #B5121B20;
}
.table>thead>tr>th, .table>tbody>tr>th, .table>tfoot>tr>th, .table>thead>tr>td, .table>tbody>tr>td, .table>tfoot>tr>td{
  border-spacing: 0;
  font-size: 40%;
  border-style: none;
  padding-top: 15px;
  padding-bottom: 15px;
  padding-right: 1em;
  padding-left: 1em;
  line-height: 1em;
}
table {
  margin: auto;
}
th {
  padding-left: 5mm;
  padding-right: 5mm;
}
.caption {
font-size: 20pt;
font-style: italic;
padding-top: 0;
}
.references {
font-size: 20px;
line-height: 90%;
}
/* Create three unequal columns that floats next to each other */
.column {
float: left;
padding: 0px;
}
.outer {
width: 33.1in;
height: calc(46.8in *  0.6325 );
-webkit-column-count: 3; /* Chrome, Safari, Opera */
-moz-column-count: 3; /* Firefox */
column-count: 3;
-webkit-column-fill: auto;
-moz-column-fill: auto;
column-fill: auto;
column-gap: 0;
padding-left: 0cm;
padding-right: 0cm;
/* -webkit-column-rule-width: 50%;
-moz-column-rule-width: 50%;
column-rule-width: 50%; */
-webkit-column-rule-style: none;
-moz-column-rule-style: none;
column-rule-style: none;
-webkit-column-rule-color: black;
-moz-column-rule-color: black;
column-rule-color: black;
background-color: #ffffff;
font-family: Rasa;
margin-top: calc(46.8in *  0.25 );
padding-top: 1em;
padding-bottom: 1em;
}
span.citation {
  color: #B5121B;
  font-weight: bold;
}
a {
text-decoration: none;
color: #B5121B;
}
#title {
font-size: 122pt;
text-align: left;
margin: 0;
line-height: 98%;
border-bottom: 0;
font-weight: normal;
}
#author {
color: #B5121B;
margin: 0;
line-height: 85%;
font-size: 1.17em;
}
#affiliation {
padding-top: 0.1em;
color: #00000080;
font-style: italic;
font-size: 25px;
margin: 0;
}
sup {
color: #B5121B;
}
.affiliation sup {
font-size: 20px;
}
.author {
text-align: left;
}
.author sup {
font-size: 30px;
}
.author_extra {
color: #B5121B;
margin: 0;
line-height: 85%;
font-size: 35px;
text-align: left;
}
.outer h1, h2, h3, h4, h5, h6 {
text-align: center;
margin: 0;
font-weight: bold;
}
.section h1 {
  text-align:center;
  padding-bottom:5px;
  background:
    linear-gradient(
      to left,
      #ffffff 1%,
      #ffffff 20%,
      #B5121B75 33%,
      #B5121B 50%,
      #B5121B75 66%,
      #ffffff 80%,
      #ffffff 99%
    )
    left
    bottom
    #ffffff
    no-repeat;
  background-size:100% 5px ;
  margin-top: 0.5em;
  margin-bottom: 0.5em;
}
.outer h2 {
text-align: center;
}
.outer p, .level2 {
color: #000000;
}
.outer ol {
padding-left: 8%;
padding-right: 8%;
text-align: left;
}
.main {
width: 33.1in;
height: calc(46.8in * 0.25);
position: absolute;
background-color: #B5121B;
color: #ffffff90;
font-family: Special Elite;
background-image: linear-gradient(#B5121B 50%,#B5121B);
}
.main strong {
color: #ffffff;
}
.main strong > sup {
color: #ffffff;
}
.main sup {
color: #ffffff90;
}
#main-img-left {
width: 10%;
left: 0.5in;
bottom: 0.2in;
position: absolute;
opacity: 1
}
#main-img-center {
width: 10%;
left: calc(33.1in * 0.45);
bottom: 0.5in;
position: absolute;
opacity: 1
}
#main-img-right {
width: 10%;
right: 0.5in;
bottom: 0.2in;
position: absolute;
opacity: 1
}
.main p {
font-size: 150px;
text-align: left;
margin: 0;
position: absolute;
top: 50%;
-ms-transform: translateY(-50%);
transform: translateY(-50%);
margin-left: 1em;
}
.fab {
color: #00000030;
font-size: 25px;
}
.twitter, i {
color: #00000030;
font-size: 35px;
text-decoration: none;
}
a.email {
text-decoration: none;
color: #00000030;
font-size: 35px;
}
.envelope {
color: #00000030;
font-size: 5px;
text-decoration: none;
}
.poster_wrap {
width: 33.1in;
height: 46.8in;
padding: 0cm;
}
.main_bottom {
width: 33.1in;
height: calc(46.8in * 0.1);
margin-top: calc(46.8in * 0.9);
position: absolute;
background-color: #B5121B;
background-image: linear-gradient(#B5121B 10%, #B5121B);
}
.section {
  padding-left: 10mm;
  padding-right: 10mm;
}
span > #tab:mytable {
  font-weight: bold;
}
.orcid img {
  width: 3%;
}
.section h4 {
    break-before: column;
}
</style>
</head>
<body>


<div class="poster_wrap">

<div class="column outer">
<div class="section">
<h1 id="title">Investigating a pseudo-horseshoe prior for the MR-Egger model</h1><br>
<h3 id="author" class="author">

<strong>Okezie Uche Ikonne</strong><sup> 1, <a class="orcid" href="https://orcid.org/0000-0002-8461-8099"><img src="https://raw.githubusercontent.com/brentthorne/posterdown/master/images/orcid.jpg"></a></sup><br>

<a class='envelope'><i class="fas fa-envelope"></i></a> <a href="mailto:o.uche-ikonne@lancaster.ac.uk" class="email">o.uche-ikonne@lancaster.ac.uk</a> <br>
  </h3>

<h5 id="author_extra", class="author_extra">
 Frank Dondelinger<sup>2</sup>
 Tom Palmer<sup>1</sup>
</h5>


<p id="affiliation" class="affiliation">
<sup>1</sup> Department of Mathematics and Statistics, Lancaster University<br> <sup>2</sup> Faculty of Health and Medicine, Lancaster University
</p>
</div>

<div id="introduction" class="section level1">
<h1>Introduction</h1>
<!-- Mendelian randomization (MR) uses genetic variants as instruments to estimate causal effects in epidemiology (@davey2003mendelian). -->
<ul>
<li>The MR-Egger model can consistently estimate the causal effect in the presence of pleiotropy given the InSIDE assumption holds (<span class="citation">Bowden, Davey Smith, and Burgess (2015)</span>).</li>
<li><span class="citation">Schmidt and Dudbridge (2017)</span> used weakly informative priors for the MR-Egger model. Other authors have investigated alternative prior distributions in MR analyses (<span class="citation">Berzuini et al. (2018)</span>).</li>
</ul>
<!--
<div class="figure" style="text-align: center">
<div id="htmlwidget-95e7d781744f1de0f8da" style="width:672px;height:350px;" class="grViz html-widget"></div>
<script type="application/json" data-for="htmlwidget-95e7d781744f1de0f8da">{"x":{"diagram":"\n      digraph mrdag {\n\n      graph [rankdir=TB, layout=neato]\n\n      node [shape=box,eight=0.3, width=0.3]\n      U [label=\"Confounders\", pos=\"3,1!\"]\n      \n      node [shape=box, height=0.3, width=0.3]\n      Y1 [label=\"Outcome\", pos=\"4,0!\"]\n\n      node [shape=box, height=0.3, width=0.3]\n      G2 [label=\"SNPs\", pos=\"0,0!\"]\n      X1 [label=\"Exposure\", pos=\"2,0!\"]\n      \n      { rank = same;  Y1 }\n\n      \n\t    G2 -> {X1}\n      U -> X1\n      U -> Y1\n      X1 -> {Y1}\n      \n      }\n      ","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>
<p class="caption">(\#fig:dag)DAG representing MR analysis</p>
</div>
-->
<p>The objectives of this research work are to:</p>
<ul>
<li>implement Bayesian estimation of IVW and MR-Egger models for a range of prior distributions in an R package.</li>
<li>investigate model performance for a range of simulated pleiotropic scenarios and a range of priors.</li>
</ul>
</div>
<div id="methods" class="section level1">
<h1>Methods</h1>
<ul>
<li><p>We implemented Bayesian estimation of the IVW and MR-Egger models in an R package <code>mrbayes</code> which automates fitting these models in the JAGS software.</p></li>
<li><p>We provide the user with a choice of priors or let them specify their own.</p></li>
<li><p>The MR-Egger model is written as;
<span class="math display">\[\frac{\Gamma_j}{\sigma_{y_j}^2} = \frac{\alpha}{{\sigma_{y_j}^2}} + \frac{\beta\gamma_j}{{\sigma_{y_j}^2}} + \varepsilon_j,\quad \varepsilon_j \sim N(0,\sigma^2)\]</span></p></li>
<li><p>Uninformative Prior <span class="math display">\[p(\alpha) \sim N(0,1000),\ p(\beta) \sim N(0,1000),\ p(\sigma) \sim U(10,10)\]</span></p></li>
<li><p>Weakly Informative Prior <span class="math display">\[p(\alpha) \sim N(0,1),\ p(\beta) \sim N(0,1),\ p(\sigma) \sim U(10,10)\]</span></p></li>
</ul>
<!-- The posterior of the models is decomposed as; $$P(\mu_\beta,\sigma_\beta|\Gamma_j) \propto P(\Gamma_j|\mu_\beta,\sigma_\beta)P(\mu_\beta,\sigma_\beta)$$ -->
<ul>
<li><p>Pseudo-Horseshoe Prior<span class="math display">\[p(\alpha) \sim N(0,1),\ p(\beta) \sim C(0,1),\ p(\sigma) \sim IG(0.5,0.5)\]</span></p></li>
<li><p>Figure <a href="#fig:dens">1</a> shows the densities of the priors.</p></li>
</ul>
<!--
$$
\begin{aligned}
  p(\alpha) &\sim N(0,1)\\
  p(\beta) &\sim C(0,1)\\
  p(\sigma) &\sim IG(0.5,0.5).
\end{aligned}
$$
-->
<!--
The posterior distribution for inference is given by,
$$
P(\alpha,\beta,\sigma|\Gamma_j) \propto P(\Gamma_j|\alpha,\beta,\sigma)P(\alpha,\beta,\sigma).
$$
-->
<!--
Implementing the horseshoe prior on a summary-level dataset would lead to overshrinkage reducing the effect of strong instruments.
-->
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<div id="simulations" class="section level2">
<h2>Simulations</h2>
<ul>
<li>We simulated two-sample summary-level data under directional pleiotropy with a true value of the causal effect of 0.05. The performance of the model was assessed using coverage and power. Results in table <a href="#tab:simtab">1</a> and figure <a href="#fig:simplot">2</a>.</li>
</ul>
<!-- Comments about results -->
</div>
<div id="example" class="section level2">
<h2>Example</h2>
<ul>
<li>We fitted summary data models to a dataset investigating the effect of body mass index on insulin resistance <span class="citation">(Richmond et al. 2017)</span>.</li>
<li>We compared Bayesian MR-Egger model estimates from models including horseshoe priors from the horseshoe package <span class="citation">(van der Pas et al. 2016)</span>.</li>
<li>Results are presented in table <a href="#tab:restab">2</a> and figure <a href="#fig:plot">3</a>.</li>
</ul>
</div>
</div>
<div id="conclusion" class="section level1">
<h1>Conclusion</h1>
<ul>
<li>We present Bayesian estimation of the IVW and MR-Egger models in our <code>mrbayes</code> package.</li>
<li>In future work we could implement Bayesian estimation of MVMR models and perform estimation using other programs such as Stan.</li>
</ul>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-berzuini2018bayesian">
<p>Berzuini, Carlo, Hui Guo, Stephen Burgess, and Luisa Bernardinelli. 2018. “A Bayesian Approach to Mendelian Randomization with Multiple Pleiotropic Variants.” <em>Biostatistics</em>.</p>
</div>
<div id="ref-bowden-ije-2015">
<p>Bowden, Jack, George Davey Smith, and Stephen Burgess. 2015. “Mendelian randomization with invalid instruments: effect estimation and bias detection through Egger regression.” <em>International Journal of Epidemiology</em> 44 (2): 512–25. <a href="https://dx.doi.org/10.1093/ije/dyv080">https://dx.doi.org/10.1093/ije/dyv080</a>.</p>
</div>
<div id="ref-richmond2017investigating">
<p>Richmond, Rebecca, Kaitlin Wade, Laura Corbin, Jack Bowden, Gibran Hemani, Nicholas Timpson, and George Davey Smith. 2017. “Investigating the role of insulin in increased adiposity: Bi-directional Mendelian randomization study.” <em>bioRxiv</em>, 155739. <a href="https://doi.org/10.1101/155739">https://doi.org/10.1101/155739</a>.</p>
</div>
<div id="ref-schmidt-ije-2017">
<p>Schmidt, A F, and F Dudbridge. 2017. “Mendelian randomization with Egger pleiotropy correction and weakly informative Bayesian priors.” <em>International Journal of Epidemiology</em> 47 (4): 1217–28. <a href="https://dx.doi.org/10.1093/ije/dyx254">https://dx.doi.org/10.1093/ije/dyx254</a>.</p>
</div>
<div id="ref-horsepackage">
<p>van der Pas, Stephanie, James Scott, Antik Chakraborty, and Anirban Bhattacharya. 2016. <em>Horseshoe: Implementation of the Horseshoe Prior</em>. <a href="https://CRAN.R-project.org/package=horseshoe">https://CRAN.R-project.org/package=horseshoe</a>.</p>
</div>
</div>
<div id="section" class="section level4">
<h4></h4>
</div>
</div>
<div id="figures-and-tables" class="section level1">
<h1>Figures and Tables</h1>
<div class="figure" style="text-align: center"><span id="fig:dens"></span>
<img src="mr2019-okezie-poster_files/figure-html/dens-1.png" alt="Density of alternative prior distributions implemented in our package." width="672" />
<p class="caption">
Figure 1: Density of alternative prior distributions implemented in our package.
</p>
</div>
<table class="table table-striped" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:simtab">Table 1: </span>Model performance under directional pleiotropy.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
IVW
</th>
<th style="text-align:right;">
MR-Egger
</th>
<th style="text-align:right;">
MR-Egger with pseudo-HS prior
</th>
<th style="text-align:right;">
MR-Egger with horseshoe prior
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Estimate
</td>
<td style="text-align:right;">
0.1607
</td>
<td style="text-align:right;">
0.0293
</td>
<td style="text-align:right;">
0.0302
</td>
<td style="text-align:right;">
0.0374
</td>
</tr>
<tr>
<td style="text-align:left;">
Power
</td>
<td style="text-align:right;">
1.0000
</td>
<td style="text-align:right;">
0.1044
</td>
<td style="text-align:right;">
0.0950
</td>
<td style="text-align:right;">
0.0994
</td>
</tr>
<tr>
<td style="text-align:left;">
Coverage
</td>
<td style="text-align:right;">
0.0036
</td>
<td style="text-align:right;">
0.8946
</td>
<td style="text-align:right;">
0.9046
</td>
<td style="text-align:right;">
0.9044
</td>
</tr>
</tbody>
</table>
<div class="figure" style="text-align: center"><span id="fig:simplot"></span>
<img src="mr2019-okezie-poster_files/figure-html/simplot-1.png" alt="Distribution of causal effect estimates under directional pleiotropy." width="672" />
<p class="caption">
Figure 2: Distribution of causal effect estimates under directional pleiotropy.
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:plot"></span>
<img src="mr2019-okezie-poster_files/figure-html/plot-1.png" alt="Scatter plot of genotype-disease versus genotype-phenotype estimates for the effect of BMI on insulin resistance." width="672" />
<p class="caption">
Figure 3: Scatter plot of genotype-disease versus genotype-phenotype estimates for the effect of BMI on insulin resistance.
</p>
</div>
<table class="table table-striped" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:restab">Table 2: </span>Estimates of the causal effect of BMI on insulin resistance.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Model
</th>
<th style="text-align:left;">
Coefficient
</th>
<th style="text-align:left;">
Estimate
</th>
<th style="text-align:left;">
95% Confidence/Credible Interval
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
IVW
</td>
<td style="text-align:left;">
Slope
</td>
<td style="text-align:left;">
0.5797
</td>
<td style="text-align:left;">
-0.1985, 1.3579
</td>
</tr>
<tr>
<td style="text-align:left;">
MR-Egger
</td>
<td style="text-align:left;">
Intercept
</td>
<td style="text-align:left;">
-0.0544
</td>
<td style="text-align:left;">
-0.1258, 04
</td>
</tr>
<tr>
<td style="text-align:left;">
MR-Egger
</td>
<td style="text-align:left;">
Slope
</td>
<td style="text-align:left;">
3.7586
</td>
<td style="text-align:left;">
-0.4793, 7.9966
</td>
</tr>
<tr>
<td style="text-align:left;">
MR-Egger with pseudo-HS prior
</td>
<td style="text-align:left;">
Intercept
</td>
<td style="text-align:left;">
-0.0143
</td>
<td style="text-align:left;">
-0.0862, 0.0327
</td>
</tr>
<tr>
<td style="text-align:left;">
MR-Egger with pseudo-HS prior
</td>
<td style="text-align:left;">
Slope
</td>
<td style="text-align:left;">
1.3488
</td>
<td style="text-align:left;">
-1.2967, 5.6133
</td>
</tr>
<tr>
<td style="text-align:left;">
MR-Egger with HS prior
</td>
<td style="text-align:left;">
Intercept
</td>
<td style="text-align:left;">
-0.023
</td>
<td style="text-align:left;">
-0.0997, 0.0248
</td>
</tr>
<tr>
<td style="text-align:left;">
MR-Egger with HS prior
</td>
<td style="text-align:left;">
Slope
</td>
<td style="text-align:left;">
1.8779
</td>
<td style="text-align:left;">
-0.9604, 64
</td>
</tr>
</tbody>
</table>
</div>

</div>
<div class="main">
<p>Bayesian estimation of the <strong>MR-Egger</strong> model using <strong>informative priors</strong> can <strong>reduce bias</strong> in the presence of <strong>pleiotropy</strong>.</p>
</div>
<div class="main_bottom">
<img id="main-img-left" src=Figures/Shield-Twitter.jpg>
<img id="main-img-center" src=>
<img id="main-img-right" src=Figures/Shield-Twitter.jpg>
</div>
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
var script = document.createElement("script");
script.type = "text/javascript";
var src = "true";
if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
if (location.protocol !== "file:" && /^https?:/.test(src))
src = src.replace(/^https?:/, '');
script.src = src;
document.getElementsByTagName("head")[0].appendChild(script);
})();
</script>


</body>
</html>
